src/module_1/module_1_meteo_api.py

```diff
@@ -1,7 +1,248 @@
+    df = pd.DataFrame(daily_data)
+    # Add the city name as a column
+    df["city"] = city
+    df["time"] = pd.to_datetime(df["time"])
+    return df
+
+
+def calculate_mean_std(df: pd.DataFrame) -> pd.DataFrame:
+    """
+    Calculates the mean and standard deviation for each prefix in the \
+    DataFrame's columns.
+    :param df: The DataFrame to calculate the mean and standard deviation for.
+    :return: A new DataFrame with the mean and standard deviation for each prefix in the 
+    original DataFrame's columns.
+    """
+    df = df.copy()
+    # Extracting prefixes (assuming they are separated by underscores)
+    prefixes = set(col.split("_")[0] for col in df.columns if "_" in col)
+    for prefix in prefixes:
+        relevant_cols = [col for col in df.columns if col.startswith(prefix)]
+        mean_values = df[relevant_cols].mean(axis=1)
+        df[prefix + "_mean"] = mean_values
+        std_values = df[relevant_cols].std(axis=1)
+        df[prefix + "_std"] = std_values
+        df = df.drop(relevant_cols, axis=1)
+    return df
+
+
+def plot_data(df: pd.DataFrame, variable: str) -> None:
+    """
+    Plots the annual mean and dispersion of a specified variable for each city in \
+    the DataFrame.
+    :param df: The DataFrame containing the data to plot.
+    :param variable: The variable to plot.
+    :return: None
+    """
+
+    # Create a mapping from short variable names to full names
+    short_to_full = {v.split("_")[0]: v for v in VARIABLES}
+
+    # Retrieve the full name of the variable
+    full_variable_name = short_to_full.get(variable, variable)
+
+    plt.figure(figsize=(12, 8))
+
+    # Create a distinct color for each city
+    colors = plt.cm.viridis(np.linspace(0, 1, len(COORDINATES)))
+
+    for idx, city in enumerate(COORDINATES.keys()):
+        city_data = df[df["city"] == city]
+        mean = city_data[f"{variable}_mean"]
+        std = city_data[f"{variable}_std"]
+        plt.plot(
+            city_data["year"],
+            mean,
+            label=city,
+            color=colors[idx],
+            marker="o",
+            linestyle="-",
+            linewidth=2,
+        )
+        plt.fill_between(
+            city_data["year"], mean - std, mean + std, color=colors[idx], alpha=0.3
+        )
+
+    plt.title(
+        f'Annual Mean and Dispersion of {full_variable_name.replace("_", " ").capitalize()}',
+        fontsize=14,
+    )
+    plt.xlabel("Year", fontsize=12)
+    plt.ylabel(full_variable_name.replace("_", " ").capitalize(), fontsize=12)
+    plt.xticks(fontsize=10)
+    plt.yticks(fontsize=10)
+    plt.grid(True)
+    plt.legend(loc="upper left", bbox_to_anchor=(1, 1), fontsize=10)
+    plt.tight_layout()
+    plt.show()
+
 
 def main():
-    raise NotImplementedError
+    list_data_cities = []
+    for city in COORDINATES.keys():
+        logging.info(f"Getting data for {city}")
+        list_data_variables = []
+        for variable in VARIABLES:
```

Comment:
The API accepts several variables and models at once. This would reduce the amount of calls and boost efficiency

Reply:
Sí, en principio, lo hacía todo de una o al menos sin tantas iteraciones. Pero como el rate limit me daba tantos problemas, al final lo he desagregado en muchas calls de poco contenido para ir viendo como lidiar con el limite de la API.